panther_driver/src/battery_node.py

```diff
@@ -6,58 +6,52 @@
 
 import rospy
 from sensor_msgs.msg import BatteryState
-from panther_msgs.msg import BatteryDriver
+from panther_msgs.msg import DriverStateArr
 
 
-class BatteryNode:
-    def __init__(self, name):
-        rospy.init_node(name)
+class BatteryNode(object):
+    def __init__(self, name) -> None:
+        rospy.init_node(name, anonymous=False)
 
-        self._battery_driv_sub = rospy.Subscriber(
-            'battery_driver', BatteryDriver, self._battery_driv_callback
-        )
+        assert rospy.has_param(
+            "~measurements_file"
+        ), f"[{rospy.get_name()}] measurements_file not defined, can not start collecting ADC measurements"
+
+        measurements_file = rospy.get_param('~measurements_file')
+        loop_rate = rospy.get_param('~loop_rate', 20)
+
+        with open(measurements_file, 'r') as stream:
+            try:
+                self.config_file = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
 
         self.V_driv_front = NaN
         self.V_driv_rear = NaN
         self.I_driv_front = NaN
         self.I_driv_rear = NaN
 
-        # Battery
+        self._battery_driv_sub = rospy.Subscriber('panther_driver/state', DriverStateArr, self._battery_driv_callback)
+
         self._battery_publisher = rospy.Publisher('battery', BatteryState, queue_size=1)
         self._battery1_publisher = rospy.Publisher('battery1', BatteryState, queue_size=1)
         self._battery2_publisher = rospy.Publisher('battery2', BatteryState, queue_size=1)
 
-        if rospy.has_param('~measurements_file'):
-            measurements_file = rospy.get_param('~measurements_file')
-        else:
-            rospy.logerr(
-                f'[{rospy.get_name()}] measurements_file not defined, can not start collecting ADC measurements'
-            )
-            return
+        rospy.Timer(rospy.Duration(1 / loop_rate), self._battery_timer_callback)
```

Comment:
```suggestion
        rospy.Timer(rospy.Duration(1.0 / loop_rate), self._battery_timer_callback)
```

Even though python3 support type casting during division and multiplication, it is better to use float values